docs/lesson-plans/week-4.md

```diff
@@ -0,0 +1,623 @@
+operation of CRUD, so it's usually used when you're trying to just get information from a server. In fact, whenever you
+load a website using a web browser, you're sending `GET` requests to the website you want to access. 
+
+### What is REST
+
+Representational State Transfer (REST) is a set of constraints for building web services. RESTful web services, or 
+web applications that conform to REST techniques require the following constraints.
+
+- *Server-Client Architecture*: The server-client architecture discussed earlier.
+- *Stateless*: the server can't store any client context on the server, so clients are the only ones that know about 
+their session state and contain all information necessary to make a request. If the server needs to know any of the 
+client context (like authentication information), it should be stored in a database.
+- *Cacheable*: Responses can be marked as cacheable or non-cacheable so that optimizations can be performed for 
+scalability and performance.
+- *Layered System*: A client shouldn't need to know if it's connected directly to the end server or if there are 
+intermediary services.
+- *Uniform Interface*: The server's resources and implementation should be decoupled from what is returned to the 
+client.
+
+### Postman and API Clients
+
+Now that you know some of the basics of web requests and the way that the web works, let's try out some HTTP requests
+and inspect the responses. API clients are programs that nicely let you send any kind of request with any information to
+whatever URLs you define. Some of the nice ones even let you define setup scripts or post request tests for ease of use
+and validation later on.
+
+In terms of being able to be run through the command line and being on almost every computer, 
+[curl](https://curl.haxx.se/) is an extremely useful and functional (but quite complicated) tool. Feel free to try it 
+out with a command like `curl https://curl.haxx.se/`, and you should see the curl webpage in HTML. If you try 
+`curl -o test.html https://curl.haxx.se/`, and then open that file in a web browser, you should see the webpage, but
+downloaded!
+
+Working entirely on the command line like that can get really annoying though, so try out one of our favorite API 
+clients: [Postman](https://www.postman.com/). In here, you can run almost any type of request, authenticate, set
+headers and body (if applicable), run scripts and tests, see the output (in HTML too!), status code, time, size, and
+returned cookies and headers! If you find that you end up using a lot of the same requests, you can also save them and 
+pull them up later on. We've used this for classes, coop, and personal projects, and hopefully you'll find this useful.
+
+## Routing on the Backend
+
+Now we'll get to setting up the individual routes for the backend of our website so that you can access different 
+resources.
+
```

Comment:
You should mention that they should have their backend code setup from week 1 and link the video for how to do that if they have not yet